echo_server.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+import socket
+
+def response_ok():
+    return u"HTTP/1.1 200 OK\r\n \
+    Content-Type: text/plain\r\n\r\n".encode('utf-8')
+
+def response_error(error_code, reason):
+    return u"HTTP/1.1 {} {}\r\n \
+    Content-Type: text/plain\r\n\r\n".format(error_code, reason).encode('utf-8')
+
+def parse_request(request):
+    first_line = request.splitlines()[0]
+    first_line = first_line.split(' ')
```

Comment:
If what you mean is 'split on whitespace' you should omit any arguments to split:

```python
>>> foo = "This is  a string  with   extra spaces"
>>> foo.split()
['This', 'is', 'a', 'string', 'with', 'extra', 'spaces']
>>> foo.split(' ')
['This', 'is', '', 'a', 'string', '', 'with', '', '', 'extra', 'spaces']
```